tests/test_gx.py

```diff
@@ -123,6 +132,17 @@ def test_that_convert_nested_columns_to_json_does_nothing_if_no_nested_columns(
         result = self.good_runner.convert_nested_columns_to_json(df, [])
         pd.testing.assert_frame_equal(result, df)
 
+    def test_get_failed_expectations_with_failed_checkpoint_result(self):
+        expected = (
```

Comment:
Can you add test for what `get_failed_expectations` returns on a successful result?

Reply:
It is not called on a successful result. I just realized I should add `assert_not_called` checks on the successful tests for `run`.

Reply:
Oooh ok thanks. I looked over the whole test file but I'm not familiar enough to GX to understand the whole process.

Reply:
Yeah, it's just this line [here](https://github.com/Sage-Bionetworks/agora-data-tools/blob/15c6ba9b4565cc4fb0fa502620934468d3993769/src/agoradatatools/gx.py#L184) in `run` that dictates this. the `CheckpointResult` object produced by GX has a `success` property. When that returns false we run `get_failed_expectations`

I added those `assert_not_called` checks in my latest commit.